packages/ckb-indexer/src/collector.ts

```diff
@@ -0,0 +1,243 @@
+import {
+  Script,
+  utils,
+  Cell,
+  BaseCellCollector,
+  ScriptWrapper,
+} from "@ckb-lumos/base";
+import { validators } from "ckb-js-toolkit";
+import {
+  AdditionalOptions,
+  CkbQueryOptions,
+  GetCellsResults,
+  HexadecimalRange,
+  Order,
+  SearchFilter,
+} from "./indexer";
+
+import { CkbIndexer, ScriptType, SearchKey } from "./indexer";
+
+export class IndexerCollector implements BaseCellCollector {
+  constructor(public indexer: CkbIndexer, public queries: CkbQueryOptions) {
+    const defaultQuery: CkbQueryOptions = {
+      lock: undefined,
+      type: undefined,
+      argsLen: -1,
+      data: "any",
+      fromBlock: undefined,
+      toBlock: undefined,
+      order: Order.asc,
+      skip: undefined,
+      outputDataLenRange: undefined,
+      outputCapacityRange: undefined,
+      sizeLimit: undefined,
```

Comment:
```suggestion
      bufferSize: undefined,
```
`sizeLimit` may cause misunderstandings

Reply:
done